From f76f654ad7a0eaee6d0aaf6b8d1c22f2117fb4d3 Mon Sep 17 00:00:00 2001
From: Ryan Volz <rvolz@mit.edu>
Date: Wed, 3 Jun 2020 16:37:04 -0400
Subject: [PATCH] Move Python package to the gnuradio subdirectory.

This is where it was previously with GNU Radio 3.7 and it avoids a name
conflict with the libiio Python bindings.
---
 grc/iio_attr_sink.block.yml       | 2 +-
 grc/iio_attr_source.block.yml     | 2 +-
 grc/iio_attr_updater.block.yml    | 2 +-
 grc/iio_device_sink.block.yml     | 2 +-
 grc/iio_device_source.block.yml   | 2 +-
 grc/iio_fmcomms2_sink.block.yml   | 2 +-
 grc/iio_fmcomms2_source.block.yml | 2 +-
 grc/iio_fmcomms5_sink.block.yml   | 2 +-
 grc/iio_fmcomms5_source.block.yml | 2 +-
 grc/iio_pluto_sink.block.yml      | 2 +-
 grc/iio_pluto_source.block.yml    | 2 +-
 python/CMakeLists.txt             | 2 +-
 swig/CMakeLists.txt               | 2 +-
 13 files changed, 13 insertions(+), 13 deletions(-)

diff --git a/grc/iio_attr_sink.block.yml b/grc/iio_attr_sink.block.yml
index 9320b49..f093511 100644
--- a/grc/iio_attr_sink.block.yml
+++ b/grc/iio_attr_sink.block.yml
@@ -44,7 +44,7 @@ inputs:
     id: attr
 
 templates:
-    imports: import iio
+    imports: from gnuradio import iio
     make: iio.attr_sink(${uri}, ${device}, ${channel}, ${attr_type}, ${output}, ${required_enable})
 
 documentation: https://wiki.analog.com/resources/tools-software/linux-software/gnuradio
diff --git a/grc/iio_attr_source.block.yml b/grc/iio_attr_source.block.yml
index d7afc7d..ec04835 100644
--- a/grc/iio_attr_source.block.yml
+++ b/grc/iio_attr_source.block.yml
@@ -77,7 +77,7 @@ outputs:
     dtype: ${ type.fcn }
 
 templates:
-    imports: import iio
+    imports: from gnuradio import iio
     make: iio.attr_source(${uri}, ${device}, ${channel}, ${attribute}, ${update_interval_ms}, ${samples_per_update}, ${type}, ${attr_type}, ${output}, ${address}, ${required_enable})
 
 documentation: https://wiki.analog.com/resources/tools-software/linux-software/gnuradio
diff --git a/grc/iio_attr_updater.block.yml b/grc/iio_attr_updater.block.yml
index 22918e5..19df8fa 100644
--- a/grc/iio_attr_updater.block.yml
+++ b/grc/iio_attr_updater.block.yml
@@ -23,7 +23,7 @@ outputs:
     id: out
 
 templates:
-    imports: import iio
+    imports: from gnuradio import iio
     make: iio.attr_updater(${attr}, ${value}, ${update_interval_ms})
     callbacks:
       - update_value(${value})
diff --git a/grc/iio_device_sink.block.yml b/grc/iio_device_sink.block.yml
index abcd504..d8fdbdc 100644
--- a/grc/iio_device_sink.block.yml
+++ b/grc/iio_device_sink.block.yml
@@ -52,7 +52,7 @@ asserts:
 - ${ len(channels) > 0 }
 
 templates:
-    imports: import iio
+    imports: from gnuradio import iio
     make: iio.device_sink(${uri}, ${device}, ${channels}, ${device_phy}, ${params}, ${buffer_size}, ${interpolation} - 1, ${cyclic})
 
 file_format: 1
diff --git a/grc/iio_device_source.block.yml b/grc/iio_device_source.block.yml
index ea6af88..5977878 100644
--- a/grc/iio_device_source.block.yml
+++ b/grc/iio_device_source.block.yml
@@ -45,7 +45,7 @@ outputs:
     optional: true
 
 templates:
-    imports: import iio
+    imports: from gnuradio import iio
     make: iio.device_source(${uri}, ${device}, ${channels}, ${device_phy}, ${params}, ${buffer_size}, ${decimation} - 1)
 
 file_format: 1
diff --git a/grc/iio_fmcomms2_sink.block.yml b/grc/iio_fmcomms2_sink.block.yml
index 4197f11..87ccb2b 100644
--- a/grc/iio_fmcomms2_sink.block.yml
+++ b/grc/iio_fmcomms2_sink.block.yml
@@ -93,7 +93,7 @@ asserts:
 - ${ ((frequency<=6000000000) and (frequency>=47000000)) }
 
 templates:
-    imports: import iio
+    imports: from gnuradio import iio
     make: iio.fmcomms2_sink_f32c(${uri}, ${frequency}, ${samplerate}, ${bandwidth}, ${tx1_en}, ${tx2_en}, ${buffer_size}, ${cyclic}, ${rf_port_select}, ${attenuation1}, ${attenuation2}, ${filter}, ${auto_filter})
     callbacks:
       - set_params(${frequency}, ${samplerate}, ${bandwidth}, ${rf_port_select}, ${attenuation1}, ${attenuation2}, ${filter}, ${auto_filter})
diff --git a/grc/iio_fmcomms2_source.block.yml b/grc/iio_fmcomms2_source.block.yml
index 7a3559e..efa5b08 100644
--- a/grc/iio_fmcomms2_source.block.yml
+++ b/grc/iio_fmcomms2_source.block.yml
@@ -124,7 +124,7 @@ asserts:
 - ${ ((frequency<=6000000000) and (frequency>=70000000)) }
 
 templates:
-    imports: import iio
+    imports: from gnuradio import iio
     make: iio.fmcomms2_source_f32c(${uri}, ${frequency}, ${samplerate}, ${bandwidth}, ${rx1_en}, ${rx2_en}, ${buffer_size}, ${quadrature}, ${rfdc}, ${bbdc}, ${gain1}, ${manual_gain1}, ${gain2}, ${manual_gain2}, ${rf_port_select}, ${filter}, ${auto_filter})
     callbacks:
       - set_params(${frequency}, ${samplerate}, ${bandwidth}, ${quadrature}, ${rfdc}, ${bbdc}, ${gain1}, ${manual_gain1}, ${gain2}, ${manual_gain2}, ${rf_port_select}, ${filter}, ${auto_filter})
diff --git a/grc/iio_fmcomms5_sink.block.yml b/grc/iio_fmcomms5_sink.block.yml
index 7d00f3d..d9fce98 100644
--- a/grc/iio_fmcomms5_sink.block.yml
+++ b/grc/iio_fmcomms5_sink.block.yml
@@ -118,7 +118,7 @@ asserts:
 - ${ ((frequency2<=6000000000) and (frequency2>=47000000)) }
 
 templates:
-    imports: import iio
+    imports: from gnuradio import iio
     make: iio.fmcomms5_sink_f32c(${uri}, ${frequency1}, ${frequency2}, ${samplerate}, ${bandwidth}, ${tx1_en}, ${tx2_en}, ${tx3_en}, ${tx4_en}, ${buffer_size}, ${cyclic}, ${rf_port_select}, ${attenuation1}, ${attenuation2}, ${attenuation3}, ${attenuation4}, ${filter})
     callbacks:
     - set_params(${frequency1}, ${frequency2}, ${samplerate}, ${bandwidth}, ${rf_port_select}, ${attenuation1}, ${attenuation2}, ${attenuation3}, ${attenuation4})
diff --git a/grc/iio_fmcomms5_source.block.yml b/grc/iio_fmcomms5_source.block.yml
index c12650d..12c5235 100644
--- a/grc/iio_fmcomms5_source.block.yml
+++ b/grc/iio_fmcomms5_source.block.yml
@@ -165,7 +165,7 @@ asserts:
 - ${ ((frequency2<=6000000000) and (frequency2>=70000000)) }
 
 templates:
-    imports: import iio
+    imports: from gnuradio import iio
     make: iio.fmcomms5_source_f32c(${uri}, ${frequency1}, ${frequency2}, ${samplerate}, ${bandwidth}, ${rx1_en}, ${rx2_en}, ${rx3_en}, ${rx4_en}, ${buffer_size}, ${quadrature}, ${rfdc}, ${bbdc}, ${gain1}, ${manual_gain1}, ${gain2}, ${manual_gain2}, ${gain3}, ${manual_gain3}, ${gain4}, ${manual_gain4}, ${rf_port_select}, ${filter})
     callbacks:
     - set_params(${frequency1}, ${frequency2}, ${samplerate}, ${bandwidth}, ${quadrature}, ${rfdc}, ${bbdc}, ${gain1}, ${manual_gain1}, ${gain2}, ${manual_gain2}, ${gain3}, ${manual_gain3}, ${gain4}, ${manual_gain4}, ${rf_port_select})
diff --git a/grc/iio_pluto_sink.block.yml b/grc/iio_pluto_sink.block.yml
index 464598c..0a4e29e 100644
--- a/grc/iio_pluto_sink.block.yml
+++ b/grc/iio_pluto_sink.block.yml
@@ -63,7 +63,7 @@ asserts:
 - ${ ((frequency<=6000000000) and (frequency>=47000000)) }
 
 templates:
-    imports: import iio
+    imports: from gnuradio import iio
     make: iio.pluto_sink(${uri}, ${frequency}, ${samplerate}, ${bandwidth}, ${buffer_size}, ${cyclic}, ${attenuation1}, ${filter}, ${auto_filter})
     callbacks:
       - set_params(${frequency}, ${samplerate}, ${bandwidth}, ${attenuation1}, ${filter}, ${auto_filter})
diff --git a/grc/iio_pluto_source.block.yml b/grc/iio_pluto_source.block.yml
index c785eac..ce63a5c 100644
--- a/grc/iio_pluto_source.block.yml
+++ b/grc/iio_pluto_source.block.yml
@@ -85,7 +85,7 @@ asserts:
 - ${ ((frequency<=6000000000) and (frequency>=70000000)) }
 
 templates:
-    imports: import iio
+    imports: from gnuradio import iio
     make: iio.pluto_source(${uri}, ${frequency}, ${samplerate}, ${bandwidth}, ${buffer_size}, ${quadrature}, ${rfdc}, ${bbdc}, ${gain1}, ${manual_gain1}, ${filter}, ${auto_filter})
     callbacks:
       - set_params(${frequency}, ${samplerate}, ${bandwidth}, ${quadrature}, ${rfdc}, ${bbdc}, ${gain1}, ${manual_gain1}, ${filter}, ${auto_filter})
diff --git a/python/CMakeLists.txt b/python/CMakeLists.txt
index 81879a7..eaa7047 100644
--- a/python/CMakeLists.txt
+++ b/python/CMakeLists.txt
@@ -32,5 +32,5 @@ GR_PYTHON_INSTALL(
     FILES
     __init__.py
     attr_updater.py
-    DESTINATION ${GR_PYTHON_DIR}/iio
+    DESTINATION ${GR_PYTHON_DIR}/gnuradio/iio
 )
diff --git a/swig/CMakeLists.txt b/swig/CMakeLists.txt
index 5792469..753246d 100644
--- a/swig/CMakeLists.txt
+++ b/swig/CMakeLists.txt
@@ -52,7 +52,7 @@ GR_SWIG_MAKE(iio_pluto_sink_swig iio_pluto_sink_swig.i)
 # Install the build swig module
 ########################################################################
 GR_SWIG_INSTALL(TARGETS iio_swig iio_pluto_source_swig iio_pluto_sink_swig
-	DESTINATION ${GR_PYTHON_DIR}/iio)
+	DESTINATION ${GR_PYTHON_DIR}/gnuradio/iio)
 
 ########################################################################
 # Install swig .i files for development
-- 
2.20.1

